src/sensors/fsp201sensor.cpp

```diff
@@ -0,0 +1,189 @@
+
+#if BNO_USE_ARVR_STABILIZATION
+    imu.enableARVRStabilizedGameRotationVector(10);
+#else
+    imu.enableGameRotationVector(10);
+#endif
+
+
+#if ENABLE_INSPECTION
+    imu.enableRawGyro(10);
+    imu.enableRawAccelerometer(10);
+    imu.enableRawMagnetometer(10);
+#endif
+
+    lastReset = 0;
+    lastData = millis();
+    working = true;
+    configured = true;
+}
+
+void FSP201Sensor::motionLoop()
+{
+    //Look for reports from the IMU
+    while (imu.dataAvailable())
+    {
+#if ENABLE_INSPECTION
+        {
+            int16_t rX = imu.getRawGyroX();
+            int16_t rY = imu.getRawGyroY();
+            int16_t rZ = imu.getRawGyroZ();
+            uint8_t rA = imu.getGyroAccuracy();
+
+            int16_t aX = imu.getRawAccelX();
+            int16_t aY = imu.getRawAccelY();
+            int16_t aZ = imu.getRawAccelZ();
+            uint8_t aA = imu.getAccelAccuracy();
+
+            int16_t mX = imu.getRawMagX();
+            int16_t mY = imu.getRawMagY();
+            int16_t mZ = imu.getRawMagZ();
+            uint8_t mA = imu.getMagAccuracy();
+
+            Network::sendInspectionRawIMUData(sensorId, rX, rY, rZ, rA, aX, aY, aZ, aA, mX, mY, mZ, mA);
+        }
+#endif
+
+        lastReset = 0;
+        lastData = millis();
+
+        if (imu.hasNewGameQuat())
+        {
+            imu.getGameQuat(quaternion.x, quaternion.y, quaternion.z, quaternion.w, calibrationAccuracy);
+            quaternion *= sensorOffset;
+
+    #if ENABLE_INSPECTION
+            {
+                Network::sendInspectionFusedIMUData(sensorId, quaternion);
+            }
+    #endif // ENABLE_INSPECTION
+
+            if (!OPTIMIZE_UPDATES || !lastQuatSent.equalsWithEpsilon(quaternion))
+            {
+                newData = true;
+                lastQuatSent = quaternion;
+            }
+        }
+
+        if (imu.hasNewAccel())
+        {
+            float v[3];
+            uint8_t acc;
+            imu.getAccel(v[0], v[1], v[2], acc);
+            Network::sendAccel(v, PACKET_ACCEL);
+        }
+        if (m_IntPin == 255 || imu.I2CTimedOut())
+            break;
+    }
+    if (lastData + 1000 < millis() && configured)
+    {
+        while(true) {
+            BNO080Error error = imu.readError();
+            if(error.error_source == 255)
+                break;
+            lastError = error;
+            m_Logger.error("BNO08X error. Severity: %d, seq: %d, src: %d, err: %d, mod: %d, code: %d",
+                error.severity, error.error_sequence_number, error.error_source, error.error, error.error_module, error.error_code);
+        }
+        statusManager.setStatus(SlimeVR::Status::IMU_ERROR, true);
+        working = false;
+        lastData = millis();
+        uint8_t rr = imu.resetReason();
+        if (rr != lastReset)
+        {
+            lastReset = rr;
+            Network::sendError(rr, this->sensorId);
+        }
+        m_Logger.error("Sensor %d doesn't respond. Last reset reason:", sensorId, lastReset);
```

Comment:
```suggestion

        m_Logger.error("Sensor %d doesn't respond. Last reset reason:", sensorId, lastReset);
```

Reply:
There shouldn't be a new line, it's logically the same block, imo.

Reply:
`}` = new line after it, that's my rules